protocols/hyprland-surface-v1.xml

```diff
@@ -0,0 +1,103 @@
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+    OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  </copyright>
+
+  <description summary="hyprland specific wl_surface extensions">
+    This protocol exposes hyprland specific wl_surface properties.
+  </description>
+
+  <interface name="hyprland_surface_manager_v1" version="1">
+    <description summary="manager for hyprland surface objects">
+      This interface allows a client to create hyprland surface objects.
+    </description>
+
+    <request name="get_hyprland_surface">
+      <description summary="create a hyprland surface object">
+        Create a hyprland surface object for the given wayland surface.
+
+        If the wl_surface already has an associated hyprland_surface_v1 object,
+        even from a different manager, creation is a protocol error.
+      </description>
+
+      <arg name="id" type="new_id" interface="hyprland_surface_v1"/>
+      <arg name="surface" type="object" interface="wl_surface"/>
+    </request>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the hyprland surface manager">
+        Destroy the surface manager.
+        This does not destroy existing surface objects.
+      </description>
+    </request>
+
+    <enum name="error">
+      <entry name="already_constructed" value="0" summary="wl_surface already has a hyprland surface object"/>
+    </enum>
+  </interface>
+
+  <interface name="hyprland_surface_v1" version="1">
+    <description summary="hyprland-specific wl_surface properties">
+      This interface allows access to hyprland-specific properties of a wl_surface.
+
+      Once the wl_surface has been destroyed, the hyprland surface object must be
+      destroyed as well. All other operations are a protocol error.
+    </description>
+
+    <request name="set_opacity">
+      <description summary="set the overall opacity of the surface">
+        Sets a multiplier for the overall opacity of the surface.
+        This multiplier applies to visual effects such as blur behind the surface
+        in addition to the surface's content.
+
+        The default value is 1.0.
+        Setting a value outside of the range 0.0 - 1.0 (inclusive) is a protocol error.
+        Does not take effect until wl_surface.commit is called.
+      </description>
+
+      <arg name="opacity" type="fixed"/>
+    </request>
+
+    <enum name="set_opacity_error">
+      <entry name="out_of_range" value="0" summary="given opacity was not in the range 0.0 - 1.0 (inclusive)"/>
+    </enum>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy the hyprland surface interface">
+        Destroy the hyprland surface object,
+        resetting any changes made to the surface to their default values.
```

Comment:
immediately resetting (no double-buffering)

Reply:
Hm. The implementation currently resets immediately but that doesn't have to be the case. Maybe we should specify that changes waiting on a commit will only happen with said commit instead. Thoughts?

Reply:
I would say pending changes are discarded and the state is immediately reverted

Reply:
Pending changes are already discarded, but the opacity reverts to 1.0. Wouldn't it be better to wait for the next commit for that?

Reply:
no, destroying an object destroys its atomic state in my mind.

Reply:
Ok we can do that.